src/components/MapView.jsx

```diff
@@ -96,19 +96,21 @@ export const MapView = () => {
   // }ks
 
   const createConnection = (coordinates, id, stopType, name, ref) => {
-    const isOsm = stopType === 0;
-    const entryPoint = {coordinates, id, isOsm, name, ref};
 
-    if (connectionData.length === 1) {
-      if (connectionData[0].isOsm === isOsm) {
-        updateConnectionInfo("Stops mustn't be the same kind!");
-        return;
-      }
-      if (connectionInfo) {
-        updateConnectionInfo(null);
+    if (connectionData.length < 2) {
+      const isOsm = stopType === 0;
+      const entryPoint = {coordinates, id, isOsm, name, ref};
+
+      if (connectionData.length === 1) {
+        if (connectionData[0].isOsm === isOsm) {
+          updateConnectionInfo("Stops mustn't be the same kind!");
```

Comment:
if this "Stops mustn't be the same kind!" is desired then 
`connectionData[0].isOsm === isOsm`
should be `!(connectionData[0].isOsm ^ isOsm)`

Reply:
Applied with adjustments